### PR TITLE
Add browserUrl option to Puppeteer.ConnectAsync

### LINKED ISSUE #869: Add browserUrl option to puppeteer.connect
https://github.com/kblok/puppeteer/pull/15/commits/15af75f9a24ad9a91ab7cd7a57290e212c8c1c49 and 
https://github.com/kblok/puppeteer/pull/15/commits/d346cb57b4e21f5b7dbff4051b45c6d1e9da68d6

### PR DESCRIPTION
**Pull request by kblok**
closes #869

### CODE REVIEW

File: lib/PuppeteerSharp/Launcher.cs [hunk 1]
```diff
@@ -100,6 +111,25 @@ public async Task<Browser> ConnectAsync(ConnectOptions options)
             }
         }
 
+        private async Task<string> GetWSEndpointAsync(string browserURL)
+        {
+            try
+            {
+                if (Uri.TryCreate(new Uri(browserURL), "/json/version", out var endpointURL))
+                {
+                    var client = new HttpClient();
```
Comment (Meir017) [inline lib/PuppeteerSharp/Launcher.cs]: shouldn't we create a static `HttpClient` or at least a single instance for the browser?

Comment (kblok) [reply lib/PuppeteerSharp/Launcher.cs]: I don’t think this part is critical. When do you hit Launcher.ConnectAsync will be call too many time so as a local HttpClient instance would be a problem?

Comment (kblok) [reply lib/PuppeteerSharp/Launcher.cs]: I don’t think this part is critical. When do you hit Launcher.ConnectAsync will be call too many time so as a local HttpClient instance would be a problem?

Comment (Meir017) [reply lib/PuppeteerSharp/Launcher.cs]: OK, I wasn't sure about this but just wanted to be sure

**Pull request merged**